binary_search_tree/tree.py

```diff
@@ -9,51 +9,201 @@ def __init__(self, key, val = None):
         self.right = None
         
 
-
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    def add_helper(self, current_node, key, value):
+        if current_node == None:
+            return TreeNode(key, value)
+        
+        if key <= current_node.key:
+            current_node.left = self.add_helper(current_node.left, key, value)
+        else:
+            current_node.right = self.add_helper(current_node.right, key, value)
+        return current_node
+    
+    # Time Complexity: O(1) constant b/c only 1 node added each time function is used
+    # Space Complexity: O(1) 
     def add(self, key, value = None):
```

Comment:
👍 But time complexity is O(log n) for a balanced tree because you need to find *where* in the tree to insert.  Same with space complexity.